packages/application-generic/src/services/distributed-lock/distributed-lock.service.ts

```diff
@@ -7,6 +7,7 @@ import {
   InMemoryProviderService,
 } from '../in-memory-provider';
 
+export const IOREDIS_AUTOPIPELINING_PREFIX = '{distributed-lock}';
```

Comment:
So one of my concerns regarding the implementation is by creating this prefix we will be only using a single node out of the cluster for the whole distributed lock mechanism vs utilizing sharding on redis.

Instead of creating a 0 cardinality prefix, we can maybe utilize the prefix using the `environmentId` that is already part of the key itself, similar to the approach we took with the caching service. Wdyt?

Reply:
Then that change would be at Cluster config. Could it affect negatively to the Cache service?

Reply:
Why at Cluster Config? I think it should be just in the code where we generate the `key` for distributed lock no?

Reply:
From ioredis test suites I understand that's the way of doing and enabling autopipelining automatically locates in the slots based on the keyPrefix assigned:
https://github.com/luin/ioredis/blob/eabb70ed835828ad85b602a217992486e05d2218/test/functional/cluster/autopipelining.ts#L188

Let's take it offline to align in the solution.